apps/backoffice-v2/src/api/auth.ts

```diff
@@ -1,53 +1,61 @@
 import { ISignInProps } from '../lib/react-query/mutations/useSignInMutation/interfaces';
+import { handleZodError } from '../utils/handle-zod-error/handle-zod-error';
+import { endpoints } from './endpoints';
+import { apiClient } from './api-client';
+import { z } from 'zod';
 
 export const auth = {
-  // signIn: async ({ callbackUrl, provider }: ISignInProps) => {
-  //   const [data, error] = await apiClient({
-  //     endpoint: endpoints.auth.signIn[provider].endpoint(),
-  //     method: endpoints.auth.signIn[provider].method,
-  //     schema: z.any(),
-  //     body: {
-  //       callbackUrl,
-  //     },
-  //     options: {
-  //       headers: {
-  //         /**
-  //          * Make sure headers like Authorization or credentials
-  //          * set in {@link apiClient} don't get in the way if the
-  //          * sign in route uses a different authentication method
-  //          * or doesn't use one.
-  //          */
-  //         // Authorization: `Bearer ${token}`,
-  //       },
-  //     },
-  //   });
-  //
-  //   return handleZodError(error, data);
-  // },
-  signIn: async ({ callbackUrl, provider }: ISignInProps) => ({}),
-  // signOut: async ({ callbackUrl }: ISignInProps) => {
-  //   const [data, error] = await apiClient({
-  //     endpoint: endpoints.auth.signOut.endpoint(),
-  //     method: endpoints.auth.signOut.method,
-  //     schema: z.any(),
-  //     body: {
-  //       callbackUrl,
-  //     },
-  //   });
-  //
-  //   return handleZodError(error, data);
-  // },
-  signOut: async ({ callbackUrl }: ISignInProps) => ({}),
+  signIn: async ({ callbackUrl, provider, body }: ISignInProps) => {
+    const [data, error] = await apiClient({
+      endpoint: endpoints.auth.signIn[provider].endpoint(),
+      method: endpoints.auth.signIn[provider].method,
+      schema: z.any(),
+      body: {
+        ...body,
+        callbackUrl,
+      },
+      options: {
+        headers: {
+          /**
+           * Make sure headers like Authorization or credentials
+           * set in {@link apiClient} don't get in the way if the
+           * sign in route uses a different authentication method
+           * or doesn't use one.
+           */
+          // Authorization: `Bearer ${token}`,
+        },
+      },
+    });
+
+    return handleZodError(error, data);
+  },
+  signOut: async ({ callbackUrl }: ISignInProps) => {
+    const [data, error] = await apiClient({
+      endpoint: endpoints.auth.signOut.endpoint(),
+      method: endpoints.auth.signOut.method,
+      schema: z.any(),
+      body: {
+        callbackUrl,
+      },
+    });
+
+    return handleZodError(error, data);
+  },
   getSession: async () => {
```

Comment:
Im not sure i understand the usage of `getSession`, can you explain?

Reply:
Both getting the information of the authenticated user, i.e. email which is displayed in the backoffice UI, and checking client-side if the user is authenticated for the redirects and deciding which layout to render.